CRM/Contact/BAO/GroupContact.php

```diff
@@ -498,6 +500,8 @@ public static function getGroupId($groupContactID) {
    * @param string $method
    */
   public static function create($params, $contactId, $ignorePermission = FALSE, $method = 'Admin') {
+    CRM_Core_Error::deprecatedFunctionWarning('Use the GroupContact API');
```

Comment:
@mattwire @eileenmcnaughton given there is a general move I think towards wanting create functions on BAO is this the right way thing thing to do or not or am I wrong on that front?

Reply:
@eileenmcnaughton @seamuslee001 I added the deprecated warning because this function is only being called from the createProfileContact() function and is not being used afterwards. But it's weird and non-standard whereas the `add` does something far more sensible. I'd be more than happy to see a further cleanup that moves the existing `add` to replace this `create` but wanted to be sure that no extensions etc. are (mis)-using the existing create first.

Reply:
@mattwire what we have done elsewhere is rename the function & 'reclaim' create but have a deprecated path through create to the old deprecated function, depending on the parameters & otherwise just call add(). 

It seems like the current code would be passing in an array of ids so it would be identifiable?

Reply:
@eileenmcnaughton @seamuslee001 Ok, how about now?